src/main/java/frc/team2412/robot/util/PathPlannerAutos.java

```diff
@@ -0,0 +1,247 @@
+package frc.team2412.robot.util;
+
+import com.pathplanner.lib.path.PathPlannerPath;
+import com.pathplanner.lib.path.PathPlannerTrajectory;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.Filesystem;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class PathPlannerAutos {
+	// Parsing utils
+
+	private static int indexOfEnd(String text, String target, int fromIndex) {
+		int i = text.indexOf(target, fromIndex);
+		if (i < 0) {
+			return i;
+		}
+		return i + target.length();
+	}
+
+	private static int skipWhitespace(String text, int i) {
+		while (Character.isWhitespace(text.charAt(i))) {
+			++i;
+		}
+		return i;
+	}
+
+	private static String readJsonString(String text, int quoteStart) {
+		int stringStart = quoteStart + 1;
+		int stringEnd = text.indexOf("\"", stringStart);
+		return text.substring(stringStart, stringEnd);
+	}
+
+	private static String readJsonValue(String text, int start) {
+		start = skipWhitespace(text, start);
+		int comma = text.indexOf(",", start);
+		int objectEnd = text.indexOf("}", start);
+		if (comma < 0) {
+			return text.substring(start, objectEnd);
+		}
+		if (objectEnd < 0) {
+			return text.substring(start, comma);
+		}
+		return text.substring(start, Math.min(comma, objectEnd));
+	}
+
+	private static boolean readJsonBoolean(String text, int start) {
+		// False for values that are not "true" or "false" (case insensitive)
+		return Boolean.parseBoolean(readJsonValue(text, start));
+	}
+
+	private static double readJsonDouble(String text, int start) throws NumberFormatException {
+		return Double.parseDouble(readJsonValue(text, start));
```

Comment:
Consider using one of the Json libraries wpilib uses internally for parsing Json instead of writing your own simple parser: https://github.com/FasterXML/jackson or https://github.com/google/gson
Not blocking for this PR, but would help make this more resilient and ideally re-use code already linked into the program.

Reply:
Oops, I was worried about adding a dependency but forgot WPILib already uses some JSON libraries. I'll switch to using FasterXML/jackson tomorrow.